.github/workflows/smoke.yml

```diff
@@ -0,0 +1,62 @@
+name: Smoke tests for AWS SAM CLI
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+  schedule:
+    - cron: '0 5 * * *' # 05:00 server time every day
+
+# Only one pull-request triggered run should be executed at a time
+# (head_ref is only set for PR events, otherwise fallback to run_id which differs for every run).
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+env:
+  SAM_CLI_TELEMETRY: "0"
+  AWS_ACCESS_KEY_ID: "test"
+  AWS_SECRET_ACCESS_KEY: "test"
+  AWS_DEFAULT_REGION: us-east-1
+  AWS_REGION: us-east-1
+  AWS_ENDPOINT_URL: http://localhost:4566
+  VENV_DIR: ".venv"
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.11"
+    - name: Start LocalStack
+      uses: LocalStack/setup-localstack@v0.1.2
+      with:
+        image-tag: 'latest'
+        install-awslocal: 'true'
+    - name: Install the package
+      run: |
+        make setup-venv
+        $VENV_DIR/bin/python -m pip install -e .
+    - name: Set up the stack
+      run: |
+        cd sample && \
+          ../$VENV_DIR/bin/samlocal deploy --no-confirm-changeset
+    - name: Make a request into the endpoint
+      run: |
+        TEST_URL="$(aws cloudformation describe-stacks --stack-name sample |  jq -r '.Stacks[0].Outputs[] | select(.OutputKey == "HelloWorldApi") | .OutputValue')" && \
+        echo "Making request against $TEST_URL" && \
+        RESULT="$(curl -v $TEST_URL)"
```

Comment:
it seems good enough here. I just wonder why we don't simply invoke the integration test with `AWS_ENDPOINT_URL` here instead (with a new botocore version) 🤔 
given it's already nicely generated ;)

but feel free to go with the easiest thing 👍

Reply:
I think this is a more realistic test as it exercises what the user would do i.e. making a request with an external HTTP client.

Reply:
fair point, both works 👍